File: app/src/main/java/line/util/ToolUtil.java
package line.util;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;
import android.widget.Toast;

import line.LineApplication;

/**
 * 工具类
 * Created by ying.fu on 2018/7/5.
 */

public class ToolUtil {
    public static int dpToPx(Context context, float dp) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, context.getResources().getDisplayMetrics());
    }


    /**
     * px转dip
     *
     * @param px
     * @return
     */
    public static int pxToDip(int px, Context context) {
        float dip = px / context.getResources().getDisplayMetrics().density + 0.5f;
        return (int) dip;
    }

    /**
     * 获取屏幕宽度
     */
    public static int getScreenWidth(Context context) {
        float dip = context.getResources().getDisplayMetrics().widthPixels;
        return (int) dip;
    }


    public static void toast(String msg) {
        Toast.makeText(LineApplication.getContext(), msg, Toast.LENGTH_SHORT).show();
    }

    public static void toastLone(String msg) {
        Toast.makeText(LineApplication.getContext(), msg, Toast.LENGTH_LONG).show();
    }
}

File: base/src/main/java/com/line/base/util/ToolUtil.kt
package com.line.base.util

import android.content.Context
import android.util.TypedValue
import android.view.View
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import android.widget.Toast
import com.line.base.LineApplication

/**
 * 工具类
 * Created by ying.fu on 2018/7/5.
 */
object ToolUtil {
    @JvmStatic
    fun dpToPx(context: Context, dp: Float): Int {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, context.resources.displayMetrics).toInt()
    }

    @JvmStatic
    fun dpToPx(dp: Float): Int {
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, LineApplication.context.resources.displayMetrics).toInt()
    }

    /**
     * px转dip
     *
     * @param px
     * @return
     */
    fun pxToDip(px: Int, context: Context): Int {
        val dip = px / context.resources.displayMetrics.density + 0.5f
        return dip.toInt()
    }

    /**
     * 获取屏幕宽度
     */
    @JvmStatic
    fun getScreenWidth(context: Context): Int {
        val dip = context.resources.displayMetrics.widthPixels.toFloat()
        return dip.toInt()
    }

    @JvmStatic
    fun toast(msg: String?) {
        Toast.makeText(LineApplication.context, msg, Toast.LENGTH_SHORT).show()
    }

    fun toastLone(msg: String?) {
        Toast.makeText(LineApplication.context, msg, Toast.LENGTH_LONG).show()
    }


    fun View.alphaShow(duration: Long = 300L) {
        if (visibility == View.VISIBLE) {
            return
        }

        visibility = View.VISIBLE
        val anim = AlphaAnimation(0f, 1f)
        anim.duration = duration
        startAnimation(anim)
    }

    fun View.alphaGone(duration: Long = 300L) {
        if (visibility != View.VISIBLE) {
            return
        }
        val anim = AlphaAnimation(1f, 0f)
        anim.duration = duration
        anim.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation?) {

            }

            override fun onAnimationEnd(animation: Animation?) {
                visibility = View.GONE
            }

            override fun onAnimationRepeat(animation: Animation?) {
            }
        })
        startAnimation(anim)
    }
}